File: src/16/16428.py
"""
16428. A/B - 3

작성자: xCrypt0r
언어: Python 3
사용 메모리: 29,380 KB
소요 시간: 72 ms
해결 날짜: 2020년 9월 14일
"""

def main():
    A, B = map(int, input().split())
    q, r = divmod(A, B)

    if B < 0 and A != 0:
        q += 1
        r -= B

    print(q, r, sep='\n')

if __name__ == '__main__':
    main()

File: src/16/16428.cs
/**
 * 16428. A/B - 3
 *
 * 작성자: xCrypt0r
 * 언어: C# 9.0 (.NET)
 * 사용 메모리: 19,052 KB
 * 소요 시간: 168 ms
 * 해결 날짜: 2021년 9월 27일
 */

using System;
using System.Numerics;

class Program
{
    static void Main()
    {
        var input = Array.ConvertAll(Console.ReadLine().Split(' '), BigInteger.Parse);
        var (A, B) = (input[0], input[1]);
        var reminder = A % B;

        if (reminder < 0)
        {
            reminder += BigInteger.Abs(B);
        }

        var quotient = (A - reminder) / B;

        Console.WriteLine(quotient);
        Console.WriteLine(reminder);
    }
}
